### ISSUE TITLE
macOS tray icon rarely doesn't work + crashes app

### ISSUE THREAD

**Issue by nift4** [AvaloniaUI/Avalonia#14578]
## Describe the bug
I start my production app, try to use tray menu normally, and it usually works most of the time. However, sometimes, clicking an item in NativeMenu crashes the app, and it appears to be related to memory corruption as sometimes the AvnMenuItem appears to be deallocated and sometimes the _item in AvnMenuItem appears to be deallocated (attached crash log for both below).

## To Reproduce

1. Start app
2. Click on tray icon
3. Click on any menu entry
4. If app did not crash, stop app and go to 1
5. See error

## Expected behavior
No crash

## Screenshots
--

## Environment 

- OS: macOS 14.3.1
- Avalonia-Version: both 11.0.6 and 11.0.8

## Additional context
Sadly wasn't able to create a minimal reproducer, I can however link to production app repo: https://github.com/ThePBone/GalaxyBudsClient/tree/19f4561e35b91c3f540ef16f0911b6768ecc7ed8
Crash log: https://gist.github.com/nift4/aa21d4b79efec5e3bf01f2acaaabb454

**Comment by maxkatz6** [comment]
Yeah, it needs a minimal repro. Since typical TrayIcon usage doesn't crash on my m1 macbook.

**Comment by nift4** [comment]
[Archiv.zip](https://github.com/AvaloniaUI/Avalonia/files/14231840/Archiv.zip)
I was able to create a repro, it seems to be related to a Timer we use to update the menu periodically. I increased timer frequency in the repro to make the crash appear more often, we don't update every 100ms in the actual app.

**Comment by timunie** [comment]
You maybe need to use DispatcherTimer here or use Dispatcher.UiThread.Post to make sure the UI is only updated on the UI Thread? I'm just guessing here as I don't have a Mac to test.

**Comment by nift4** [comment]
![grafik](https://github.com/AvaloniaUI/Avalonia/assets/36458596/07bf91c0-3e9e-4061-944e-3e3d3f4f2489)
it's like this in the repro, do I need to fix anything else?

**Comment by timunie** [comment]
🤔 I still would try DispatcherTimer here just in case. https://learn.microsoft.com/en-us/dotnet/api/system.windows.threading.dispatchertimer?view=windowsdesktop-8.0

**Comment by nift4** [comment]
any chance to get "needs-repro" removed to increase the likelyhood of someone checking this out?

**Comment by timunie** [comment]
@nift4 could you give DispatcherTimer a go? Also worth checking nightly builds when testing. 

If you did above in your sample attached and issue persists, I'll remove that label. I've no own mac, so can't do it on my own. 

In case you need a fix quick: Also you could try to debug it on your own if you add the sample into src/samples/Sandbox. Make sure you have read build instructions.
